grpc-server/tinkerbell.go

```diff
@@ -2,47 +2,219 @@ package grpcserver
 
 import (
 	"context"
+	"fmt"
+	"log"
+	"time"
 
-	exec "github.com/tinkerbell/tink/executor"
+	"github.com/tinkerbell/tink/db"
 	pb "github.com/tinkerbell/tink/protos/workflow"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
+var workflowData = make(map[string]int)
+
 // GetWorkflowContexts implements tinkerbell.GetWorkflowContexts
 func (s *server) GetWorkflowContexts(req *pb.WorkflowContextRequest, stream pb.WorkflowSvc_GetWorkflowContextsServer) error {
-	return exec.GetWorkflowContexts(req, stream, s.db)
+	wfs, err := getWorkflowsForWorker(s.db, req.WorkerId)
+	if err != nil {
+		return err
+	}
+	for _, wf := range wfs {
+		wfContext, err := s.db.GetWorkflowContexts(context.Background(), wf)
+		if err != nil {
+			return status.Errorf(codes.Aborted, "invalid workflow %s found for worker %s", wf, req.WorkerId)
```

Comment:
Error Statement seems to be incorrect. Is it always `invalid workflow` if `err!=nil`. It could also mean sql query failed or connection to db ?

Reply:
Yeah, I agree. We should instead return the error as is to the caller.